arangod/Replication/Syncer.cpp

```diff
@@ -130,20 +138,43 @@ arangodb::Result applyCollectionDumpMarkerInternal(
     options.indexOperationMode = arangodb::Index::OperationMode::internal;
 
     try {
-      // try insert first
-      OperationResult opRes = trx.insert(coll->name(), slice, options);
+      OperationResult opRes;
+      bool useReplace = false;
+      VPackSlice keySlice = arangodb::transaction::helpers::extractKeyFromDocument(slice);
```

Comment:
is it safe to utilize this on documents received from network  / parsed from JSON ?

Reply:
Yes, it is safe. If `extractKeyFromDocument` cannot find the key at the expected position, it will perform a regular `Slice::get()` operation.